src/v/cloud_storage/access_time_tracker.cc

```diff
@@ -78,6 +78,16 @@ void access_time_tracker::remove_timestamp(std::string_view key) noexcept {
     }
 }
 
+void access_time_tracker::remove_others(const access_time_tracker& t) {
+    table_t tmp;
+    for (auto it : _table.data) {
+        if (t._table.data.contains(it.first)) {
+            tmp.data.insert(it);
```

Comment:
std::move(it) or make `it` be `const auto&` to reduce copies here? better you could probably use the table extraction API to move the literal tree nodes between tree instances since you are going to overwrite _table anyway.

Reply:
done

Reply:
it doesn't look like this was done